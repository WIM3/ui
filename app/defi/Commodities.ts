import { Commodity, CommodityId } from "./types";

export const COMMODITIES: { [key in CommodityId]: Commodity } = {
  [CommodityId.gold]: {
    id: CommodityId.gold,
    icon: "/commodities/gold.svg",
    symbol: "XAU",
    name: "Gold",
  },
};
