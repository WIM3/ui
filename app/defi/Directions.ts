export enum Directions {
  Short = "short",
  Long = "long",
}
