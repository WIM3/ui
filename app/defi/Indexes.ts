import { Index, IndexId } from "./types";

export const INDEXES: { [key in IndexId]: Index } = {
  [IndexId.GOOG]: {
    id: IndexId.GOOG,
    icon: "/indexes/google.svg",
    symbol: "GOOG",
    name: "Google",
  },
  [IndexId.ACRCA00]: {
    id: IndexId.ACRCA00,
    icon: "/indexes/sp500.svg", // TODO
    symbol: "ACRCA00",
    name: "ACRCA00",
  },
  [IndexId.BATCH04]: {
    id: IndexId.BATCH04,
    icon: "/indexes/sp500.svg", // TODO
    symbol: "BATCH04",
    name: "BATCH04",
  },
  [IndexId.BATCP04]: {
    id: IndexId.BATCP04,
    icon: "/indexes/sp500.svg", // TODO
    symbol: "BATCP04",
    name: "BATCP04",
  },
  [IndexId.CNCAD00]: {
    id: IndexId.CNCAD00,
    icon: "/indexes/sp500.svg", // TODO
    symbol: "CNCAD00",
    name: "CNCAD00",
  },
  [IndexId.SPBTC]: {
    id: IndexId.SPBTC,
    icon: "/indexes/bitcoin.svg", // TODO
    symbol: "SPBTC",
    name: "Bitcoin Index",
  },
  [IndexId.SPETH]: {
    id: IndexId.SPETH,
    icon: "/indexes/eth-mainnet.svg", // TODO
    symbol: "SPETH",
    name: "Ethereum Index",
  },
};
