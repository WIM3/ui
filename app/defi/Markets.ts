import { Market, MarketId } from "./types";

export const MARKETS: {
  [marketId in MarketId]: Market;
} = {
  [MarketId.crypto]: {
    id: MarketId.crypto,
    name: "Crypto",
    symbol: "Crypto",
    icon: "/markets/crypto.svg",
  },
  [MarketId.sp500]: {
    id: MarketId.sp500,
    name: "S&P500",
    symbol: "S&P500",
    icon: "/markets/sp500.svg",
  },
  [MarketId.SPPlatts]: {
    id: MarketId.SPPlatts,
    name: "S&P Platts",
    symbol: "S&P Platts",
    icon: "/markets/sp500.svg",
  },
  [MarketId.SPIndices]: {
    id: MarketId.SPIndices,
    name: "S&P Indices",
    symbol: "S&P Indices",
    icon: "/markets/sp500.svg",
  },
  [MarketId.commodities]: {
    id: MarketId.commodities,
    name: "Commodities",
    symbol: "Commodities",
    icon: "/markets/commodities.svg",
  },
};
