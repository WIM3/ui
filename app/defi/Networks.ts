import { Network, NetworkId, TokenId } from "./types";

export const NETWORKS: {
  [networkId in NetworkId]: Network;
} = {
  [NetworkId.ethereum]: {
    name: "Ethereum",
    symbol: "Ethereum",
    rpcUrl: process.env.RPC_URL_1!,
    infoPageUrl: "https://etherscan.io/tx/",
    infoPage: "Etherscan",
    icon: "/networks/mainnet.svg",
    defaultTokenSymbol: "ETH",
    publicRpcUrl: "",
    nativeToken: TokenId.eth,
  },
  [NetworkId.arbitrum]: {
    name: "Arbitrum",
    symbol: "Arbitrum",
    rpcUrl: process.env.RPC_URL_42161!,
    infoPageUrl: "https://arbiscan.io/tx/",
    infoPage: "Arbiscan",
    icon: "/networks/arbitrum.svg",
    defaultTokenSymbol: "ETH",
    publicRpcUrl: "https://arb1.arbitrum.io/rpc",
    nativeToken: TokenId.eth,
  },
  [NetworkId.polygon]: {
    name: "Polygon",
    symbol: "Polygon",
    rpcUrl: process.env.RPC_URL_137!,
    infoPageUrl: "https://polygonscan.com/tx/",
    infoPage: "Polygonscan",
    icon: "/networks/polygon.svg",
    defaultTokenSymbol: "MATIC",
    publicRpcUrl: "https://rpc-mainnet.maticvigil.com/",
    nativeToken: TokenId.eth,
  },
  [NetworkId.avalancheTestnet]: {
    name: "Avalanche Fuji",
    symbol: "Avalanche Testnet",
    rpcUrl: process.env.RPC_URL_43113!,
    infoPageUrl: "https://cchain.explorer.avax.network/tx/",
    infoPage: "Avax Scan",
    icon: "/networks/avalanche.svg",
    defaultTokenSymbol: "AVAX",
    publicRpcUrl: "https://api.avax-test.network/ext/bc/C/rpc",
    nativeToken: TokenId.avax,
    etherscanLink: "https://testnet.snowtrace.io/",
    secondsPerBlock: 2.82,
  },
  [NetworkId.avalanche]: {
    name: "Avalanche",
    symbol: "Avalanche",
    rpcUrl: process.env.RPC_URL_43114!,
    infoPageUrl: "https://cchain.explorer.avax.network/tx/",
    infoPage: "Avax Scan",
    icon: "/networks/avalanche.svg",
    defaultTokenSymbol: "AVAX",
    publicRpcUrl: "https://api.avax.network/ext/bc/C/rpc",
    nativeToken: TokenId.avax,
  },
  [NetworkId.moonriver]: {
    name: "Moonriver",
    symbol: "Moonriver",
    rpcUrl: process.env.RPC_URL_1285!,
    infoPageUrl: "https://blockscout.moonriver.moonbeam.network/tx/",
    infoPage: "Moonriver Blockscout",
    icon: "/networks/moonriver.svg",
    defaultTokenSymbol: "MOVR",
    publicRpcUrl: "https://rpc.moonriver.moonbeam.network",
    nativeToken: TokenId.eth,
  },
  [NetworkId.fantom]: {
    name: "Fantom",
    symbol: "Fantom",
    rpcUrl: process.env.RPC_URL_250!,
    infoPageUrl: "https://ftmscan.com/tx/",
    infoPage: "Fantom Scan",
    icon: "/networks/fantom.svg",
    defaultTokenSymbol: "FTM",
    publicRpcUrl: "https://rpc.ftm.tools",
    nativeToken: TokenId.ftm,
  },
};
