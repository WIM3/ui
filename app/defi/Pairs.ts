import { CommodityId, IndexId, MarketId, Pair, PairId, TokenId } from "./types";

export const PAIRS: { [key in PairId]: Pair } = {
  [PairId.btcusdc]: {
    id: PairId.btcusdc,
    marketId: MarketId.crypto,
    productIds: [TokenId.btc, TokenId.usdc],
  },
  [PairId.ethusdc]: {
    id: PairId.ethusdc,
    marketId: MarketId.crypto,
    productIds: [TokenId.eth, TokenId.usdc],
  },
  [PairId.avaxusdc]: {
    id: PairId.avaxusdc,
    marketId: MarketId.crypto,
    productIds: [TokenId.avax, TokenId.usdc],
  },
  [PairId.chaosusdc]: {
    id: PairId.chaosusdc,
    marketId: MarketId.crypto,
    productIds: [TokenId.chaos, TokenId.usdc],
  },
  [PairId.ftmusdc]: {
    id: PairId.ftmusdc,
    marketId: MarketId.crypto,
    productIds: [TokenId.ftm, TokenId.usdc],
  },
  [PairId.movrusdc]: {
    id: PairId.movrusdc,
    marketId: MarketId.crypto,
    productIds: [TokenId.movr, TokenId.usdc],
  },
  [PairId.googusdc]: {
    id: PairId.googusdc,
    marketId: MarketId.commodities,
    productIds: [IndexId.goog, TokenId.usdc],
  },
  [PairId.xauusdc]: {
    id: PairId.xauusdc,
    marketId: MarketId.sp500,
    productIds: [CommodityId.gold, TokenId.usdc],
  },
};
