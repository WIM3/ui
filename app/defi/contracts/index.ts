import { Contract, providers } from "ethers";

import {
  BasicTokenWithMint,
  ClearingHouse,
  BasicTokenWithMint__factory,
  ClearingHouse__factory,
} from "./types";

export const getTokenContract = (
  signer: providers.JsonRpcSigner,
  address?: string
) =>
  new Contract(
    address || process.env.USDC!,
    BasicTokenWithMint__factory.abi,
    signer
  ) as BasicTokenWithMint;

export const getClearingHouseContract = (
  signer: providers.JsonRpcSigner,
  address?: string
) =>
  new Contract(
    address || process.env.CLEARING_HOUSE!,
    ClearingHouse__factory.abi,
    signer
  ) as ClearingHouse;
