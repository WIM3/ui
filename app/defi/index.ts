export * from "./types";
export * from "./Tokens";
export * from "./Networks";
export * from "./utils";
