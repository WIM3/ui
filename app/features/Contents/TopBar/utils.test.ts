import { MarketId } from "@/defi";
import {
  generateMarketDropdownProps,
  generatePairDropdownProps,
} from "./utils";

describe("TopBar utils", () => {
  describe("generateMarketDropdownProps", () => {
    test("should generate market dropdown props", () => {
      const marketProps = generateMarketDropdownProps();
      expect(marketProps).toMatchSnapshot();
    });
  });

  describe("generateMarketDropdownProps", () => {
    test("should generate crypto market dropdown props", () => {
      const pairProps = generatePairDropdownProps();
      expect(pairProps[MarketId.crypto]).toMatchSnapshot();
    });

    test("should generate commodities market dropdown props", () => {
      const pairProps = generatePairDropdownProps();
      expect(pairProps[MarketId.commodities]).toMatchSnapshot();
    });

    test("should generate S&P 500 market dropdown props", () => {
      const pairProps = generatePairDropdownProps();
      expect(pairProps[MarketId.sp500]).toMatchSnapshot();
    });
  });
});
