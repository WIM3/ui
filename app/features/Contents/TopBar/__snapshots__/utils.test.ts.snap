// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar utils generateMarketDropdownProps should generate S&P 500 market dropdown props 1`] = `
Object {
  "options": Array [
    Object {
      "productIds": Array [
        "gold",
        "usdc",
      ],
      "value": "xausdc",
    },
  ],
  "searchable": true,
}
`;

exports[`TopBar utils generateMarketDropdownProps should generate commodities market dropdown props 1`] = `
Object {
  "options": Array [
    Object {
      "productIds": Array [
        "goog",
        "usdc",
      ],
      "value": "googusdc",
    },
  ],
  "searchable": true,
}
`;

exports[`TopBar utils generateMarketDropdownProps should generate crypto market dropdown props 1`] = `
Object {
  "options": Array [
    Object {
      "productIds": Array [
        "btc",
        "usdc",
      ],
      "value": "btcusdc",
    },
    Object {
      "productIds": Array [
        "eth",
        "usdc",
      ],
      "value": "ethusdc",
    },
    Object {
      "productIds": Array [
        "avax",
        "usdc",
      ],
      "value": "avaxusdc",
    },
    Object {
      "productIds": Array [
        "chaos",
        "usdc",
      ],
      "value": "chaosusdc",
    },
    Object {
      "productIds": Array [
        "ftm",
        "usdc",
      ],
      "value": "ftmusdc",
    },
    Object {
      "productIds": Array [
        "movr",
        "usdc",
      ],
      "value": "movrusdc",
    },
    Object {
      "productIds": Array [
        "solana",
        "usdc",
      ],
      "value": "solsusdc",
    },
  ],
  "searchable": true,
}
`;

exports[`TopBar utils generateMarketDropdownProps should generate market dropdown props 1`] = `
Object {
  "options": Array [
    Object {
      "productIds": Array [
        "crypto",
      ],
      "value": "crypto",
    },
    Object {
      "productIds": Array [
        "SPPlatts",
      ],
      "value": "SPPlatts",
    },
    Object {
      "productIds": Array [
        "SPIndices",
      ],
      "value": "SPIndices",
    },
  ],
  "searchable": true,
}
`;
